server/drivers/g15.c

```diff
@@ -75,41 +88,45 @@ MODULE_EXPORT int g15_init (Driver *drvthis)
 	p = (PrivateData *) calloc(1, sizeof(PrivateData));
 	if (p == NULL)
 		return -1;
-	if (drvthis->store_private_ptr(drvthis, p))
+
+	p->g15screen_fd = -1;
+
+	if (drvthis->store_private_ptr(drvthis, p)) {
```

Comment:
I was just moving the existing code around, but I agree that directly setting the pointer is much better, so  since I'm touching the code anyways I will change it to use a direct assignment for the pointer in the next version.